core/services/gateway/handlers/workflow/handler.go

```diff
@@ -0,0 +1,91 @@
+package workflow
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"go.uber.org/multierr"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/api"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/config"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers"
+)
+
+const (
+	MethodAddWorkflow = "add_workflow"
+	// NOTE: more methods will go here: CRUD for workflow specs; HTTP trigger/action/target; etc.
+)
+
+type workflowHandler struct {
+	donConfig      *config.DONConfig
+	don            handlers.DON
+	savedCallbacks map[string]*savedCallback
+	mu             sync.Mutex
+	lggr           logger.Logger
+}
+
+type savedCallback struct {
+	id         string
+	callbackCh chan<- handlers.UserCallbackPayload
+}
+
+var _ handlers.Handler = (*workflowHandler)(nil)
+
+func NewWorkflowHandler(donConfig *config.DONConfig, don handlers.DON, lggr logger.Logger) (*workflowHandler, error) {
+	lggr.Debugw("-------HNewWorkflowHandler")
```

Comment:
Stray debug log

Reply:
We can't have debug logs until final review?

Reply:
You can, I'm just calling it out in case it had gotten forgotten. Seemed like you removed some others.